knife/lib/chef/knife/ssh.rb

```diff
@@ -383,13 +395,11 @@ def open_session(subsession, command, pty = false)
                   ichannel.send_data("#{get_password}\n")
                 end
               end
-
               ch.on_extended_data do |_, _type, data|
-                raise ArgumentError if data.eql?("sudo: no tty present and no askpass program specified\n")
+                raise ArgumentError, "No PTY present. If a PTY is required use --require-pty" if data.eql?("sudo: no tty present and no askpass program specified\n")
```

Comment:
Will we always get this response if we fail to obtain a PTY  after requesting it, and continue with the connection ( `config[:require_pty] == false`)?